File: evals/vitabench-01-constraint-filter/filter.ts
import restaurantData from './restaurants.json';

export interface Restaurant {
  id: string;
  name: string;
  cuisine: string;
  rating: number;
  price_range: string;
  avg_price_per_person: number;
  dietary_options: string[];
  delivery_time_minutes: number;
  has_dine_in: boolean;
  is_chain: boolean;
  distance_km: number;
  business_hours: { open: string; close: string };
  closed_days: string[];
}

export interface FilterCriteria {
  /** Only include restaurants with these cuisines */
  cuisines?: string[];
  /** Minimum rating (inclusive) */
  minRating?: number;
  /** Maximum average price per person (inclusive) */
  maxPricePerPerson?: number;
  /** Restaurant must support ALL of these dietary options */
  requiredDietaryOptions?: string[];
  /** Maximum delivery time in minutes (inclusive) */
  maxDeliveryTime?: number;
  /** Must have dine-in option */
  requireDineIn?: boolean;
  /** Must be a chain restaurant */
  requireChain?: boolean;
  /** Maximum distance in km (inclusive) */
  maxDistance?: number;
  /** Restaurant must be open at this time (HH:MM format, 24hr) */
  openAt?: string;
  /** Restaurant must NOT be closed on this day (lowercase day name) */
  openOnDay?: string;
  /** Exclude these cuisine types */
  excludeCuisines?: string[];
}

/**
 * Load all restaurants from the JSON data.
 */
export function getAllRestaurants(): Restaurant[] {
  // TODO: implement
  throw new Error('Not implemented');
}

/**
 * Filter restaurants based on multiple criteria.
 * All specified criteria must be satisfied (AND logic).
 * Return results sorted by rating (highest first), then by distance (closest first) for ties.
 */
export function filterRestaurants(criteria: FilterCriteria): Restaurant[] {
  // TODO: implement
  throw new Error('Not implemented');
}

/**
 * Given a time (HH:MM, 24hr), day of week (lowercase), and a budget per person,
 * find the top N restaurants that are:
 *  - open at that time and on that day
 *  - within budget
 *  - can deliver within maxDeliveryMinutes (if specified)
 * Sorted by rating (highest first).
 */
export function findBestAvailable(
  time: string,
  dayOfWeek: string,
  budgetPerPerson: number,
  maxDeliveryMinutes?: number,
  topN?: number
): Restaurant[] {
  // TODO: implement
  throw new Error('Not implemented');
}

/**
 * Find pairs of restaurants suitable for "lunch + dinner" on a given day.
 * Lunch: must be open at lunchTime, dinner: must be open at dinnerTime.
 * Both must be open on the specified day.
 * They must be different cuisines (no same cuisine for both meals).
 * Total combined avg_price_per_person must not exceed totalBudget.
 * Return array of [lunchRestaurant, dinnerRestaurant] pairs, sorted by combined rating (highest first).
 */
export function findMealPairs(
  dayOfWeek: string,
  lunchTime: string,
  dinnerTime: string,
  totalBudget: number
): [Restaurant, Restaurant][] {
  // TODO: implement
  throw new Error('Not implemented');
}

File: evals/vitabench-01-constraint-filter/filter.py
import json
import os

_dir = os.path.dirname(os.path.abspath(__file__))
with open(os.path.join(_dir, "restaurants.json")) as _f:
    restaurant_data = json.load(_f)


def get_all_restaurants() -> list[dict]:
    """
    Load all restaurants from the JSON data.
    """
    # TODO: implement
    raise NotImplementedError("Not implemented")


def filter_restaurants(criteria: dict) -> list[dict]:
    """
    Filter restaurants based on multiple criteria.
    All specified criteria must be satisfied (AND logic).
    Return results sorted by rating (highest first), then by distance (closest first) for ties.

    criteria keys (all optional):
        cuisines (list[str]): restaurant's cuisine must be in this list
        min_rating (float): minimum rating (inclusive)
        max_price_per_person (float): maximum avg price per person (inclusive)
        required_dietary_options (list[str]): restaurant must support ALL listed options
        max_delivery_time (int): maximum delivery time in minutes (inclusive)
        require_dine_in (bool): must have dine-in option
        require_chain (bool): must be a chain restaurant
        max_distance (float): maximum distance in km (inclusive)
        open_at (str): time "HH:MM" - restaurant must be open at this time
        open_on_day (str): lowercase day name - restaurant must NOT be closed on this day
        exclude_cuisines (list[str]): restaurant's cuisine must NOT be in this list
    """
    # TODO: implement
    raise NotImplementedError("Not implemented")


def find_best_available(
    time: str,
    day_of_week: str,
    budget_per_person: float,
    max_delivery_minutes: int | None = None,
    top_n: int | None = None,
) -> list[dict]:
    """
    Given a time (HH:MM, 24hr), day of week (lowercase), and a budget per person,
    find the top N restaurants that are:
     - open at that time and on that day
     - within budget
     - can deliver within max_delivery_minutes (if specified)
    Sorted by rating (highest first).
    """
    # TODO: implement
    raise NotImplementedError("Not implemented")


def find_meal_pairs(
    day_of_week: str,
    lunch_time: str,
    dinner_time: str,
    total_budget: float,
) -> list[tuple[dict, dict]]:
    """
    Find pairs of restaurants suitable for "lunch + dinner" on a given day.
    Lunch: must be open at lunch_time, dinner: must be open at dinner_time.
    Both must be open on the specified day.
    They must be different cuisines (no same cuisine for both meals).
    Total combined avg_price_per_person must not exceed total_budget.
    Return list of (lunch_restaurant, dinner_restaurant) tuples,
    sorted by combined rating (highest first).
    """
    # TODO: implement
    raise NotImplementedError("Not implemented")
